==== ntpd/ntp_proto.c ====
2010-09-30 00:43:09-04:00, stenn@deacon.udel.edu +10 -3
  [Bug 1554]: peer may stay selected as system peer after becoming
  unreachable.

--- 1.303/ntpd/ntp_proto.c	2010-09-25 10:39:41 -07:00
+++ 1.304/ntpd/ntp_proto.c	2010-09-29 21:43:09 -07:00
@@ -2187,8 +2187,10 @@ clock_filter(
 	 * save the offset, delay and jitter. Note the jitter must not
 	 * be less than the precision.
 	 */
-	if (m == 0)
+	if (m == 0) {
+		clock_select();
 		return;
+	}
 
 	etemp = fabs(peer->offset - peer->filter_offset[k]);
 	peer->offset = peer->filter_offset[k];
